sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business.service.ts

```diff
@@ -0,0 +1,128 @@
+import { Injectable } from "@nestjs/common";
+import { LoggerService, InjectLogger } from "@sims/utilities/logger";
+import {
+  ApplicationBulkWithdrawalHeader,
+  ApplicationWithdrawalTextValidationResult,
+} from "./application-bulk-withdrawal-import-text.models";
+import { DataSource, In } from "typeorm";
+import { Application, RecordDataModelService } from "@sims/sims-db";
+import {
+  ApplicationBulkWithdrawalImportBusinessValidationModel,
+  ApplicationData,
+} from "./application-bulk-withdrawal-import-business-validation.models";
+
+type ApplicationSINDataMap = Record<string, ApplicationData>;
+/**
+ * Handles the application withdrawal bulk insert preparation.
+ */
+@Injectable()
+export class ApplicationBulkWithdrawalImportBusinessService extends RecordDataModelService<Application> {
+  constructor(private readonly dataSource: DataSource) {
+    super(dataSource.getRepository(Application));
+  }
+
+  /**
+   * Generates the application bulk withdrawal business validation model.
+   * @param textValidations text validation models to be converted to the business validation models.
+   * @param textHeader text header model.
+   * @param institutionId institution id.
+   * @param allowedLocationIds set of allowed location ids for the given institution.
+   * @returns application bulk withdrawal business models to be validated and persisted.
+   */
+  async generateBusinessValidationModels(
+    textValidations: ApplicationWithdrawalTextValidationResult[],
+    textHeader: ApplicationBulkWithdrawalHeader,
+    institutionId: number,
+    allowedLocationIds: number[],
+  ): Promise<ApplicationBulkWithdrawalImportBusinessValidationModel[]> {
+    const applicationSINDataMap: ApplicationSINDataMap = {};
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationSINData = await this.getSINForApplications(
+      applicationNumbers,
+      institutionId,
+    );
+    applicationSINData.forEach((record) => {
+      applicationSINDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
```

Comment:
Withdrawal is one of the types of scholastic standings. We should be probably checking by "Student withdrew from program". Maybe we can restrict the SQL query to return only the specific type.

Reply:
But the application could have been withdrawn for any of those 4 reasons and if it was withdrawn earlier for any reason, it would be the same behavior, right? Please let me know if I am missing something.

Reply:
The application can be archived for any of the cases but, as far as I know, withdrawal is one of the cases of the Scholastic Standing. For instance, one of the types of Scholastic Standing is called "Early Completion" which is not a withdrawal at all.
Please let me know if I am missing something or please confirm with @JasonCTang or @HRAGANBC as needed.

Reply:
Updated as per today's discussion.